src/store/modules/order/actions.ts

```diff
@@ -36,6 +37,77 @@ const actions: ActionTree<OrderState, RootState> = {
       showToast(translate("Something went wrong"));
     }
     return resp;
+  },
+  async getOrderDetails({ commit }, orderId) {
+    let resp;
+
+    const payload = {
+      "json": {
+        "params": {
+          "group": true,
+          "group.field": "orderId",
+          "group.limit": 10000,
+          "q.op": "AND"
+        } as any,
+        "query": "*:*",
+        "filter": `docType: ORDER AND orderTypeId: SALES_ORDER AND orderId: ${orderId}`
+      }
+    }
+
+    try {
+      resp = await OrderService.findOrderDetails(payload);
+
+      if (resp.status == 200 && !hasError(resp)) {
+        const order: Order = {
+          orderId: '',
+          orderName: '',
+          customer: {},
+          /** An array containing the items purchased in this order */
+          items: [],
+          statusId: '',
+          identifications: {},
+          notes: []
+        }
+
+        const orderName = process.env.VUE_APP_ORD_IDENT_TYPE_NAME
+        const orderId = process.env.VUE_APP_ORD_IDENT_TYPE_ID
+        const orderNo = process.env.VUE_APP_ORD_IDENT_TYPE_NO
+
+        resp.data.grouped.orderId.groups.map((group: any) => {
```

Comment:
Instead of iterating with map(), can we just get the first item in groups and populate values

Reply:
Sir, then can we use somthing like
```
const items = resp.data.grouped.orderId.groups[0].doclist.docs
order.orderId = group[0].orderId
...
```

Reply:
Yes, check the length of resp.data.grouped.orderId.groups and store first item in a variable. Use it further to extract data in order object

Reply:
Sir updated the code.